### DISCUSSION TITLE
Allow update entities to update report progress with decimals

### DISCUSSION THREAD

**Discussion by emontnemery** [home-assistant/architecture#1157] [Entity Models] [answered]
### Background:

Updates are slow for some devices. One example is devices communicating via low bandwidth networks such as Zigbee which need quite a bit of time to transfer the firmware.

The update entities only supports reporting an integer update progress. This means that is can take quite a few minutes before an update progress updates on the Zigbee end. This makes the user think the update isn't working, or that the update has stalled.

To solve this problem, allow float values for the progress, with a max of 2-digit precision. For example, 0.15 %.

### Proposal:

Extend the type of the `in_progress` property from `bool | int | None` to `bool | float | int | None`
Add a new property `display_precision: int | None`, with a default value of `None` to influence how the progress is displayed. If `display_precision` is `None`, the progress should be displayed without decimals.

**Comment by frenck** [comment, marked as answer]
This has been approved in the Core architectural meeting ✅ 

../Frenck
